server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -229,9 +229,9 @@ public DateTime(DateFormatter formatter, ZoneId timeZone, DateFieldMapper.Resolu
         }
 
         private DateTime(DateFormatter formatter, ZoneId timeZone, DateFieldMapper.Resolution resolution, boolean formatSortValues) {
-            this.formatter = formatter;
             this.timeZone = Objects.requireNonNull(timeZone);
-            this.parser = formatter.toDateMathParser();
+            this.formatter = formatter.withZone(timeZone);
+            this.parser = this.formatter.toDateMathParser();
```

Comment:
That's a good one :smile:

Reply:
Credit to Przemyslaw who spotted it.  I was just going to force it to UTC.